modular_citadel/code/modules/fish/fishtank.dm

```diff
@@ -0,0 +1,676 @@
+	density = 0					// Small enough to not block stuff
+	anchored = 0				// Small enough to move even when filled
+	pass_flags = PASSTABLE | LETPASSTHROW // Just like at the county fair, you can't seem to throw the ball in to win the goldfish, and it's small enough to pull onto a table
+	resistance_flags = ACID_PROOF
+
+	tank_type = "bowl"
+	water_capacity = 80			// Not very big, therefore it can't hold much. About a bucket.
+	max_fish = 1				// What a lonely fish
+
+	has_lid = 0
+	max_integrity = 15				// Not very sturdy
+	shard_count = 0				// No salvageable shards
+
+/obj/machinery/fishtank/tank
+	name = "fish tank"
+	desc = "A large glass tank designed to house aquatic creatures. Contains an integrated water circulation system."
+	icon_state = "tank1"
+	density = 1
+	anchored = 1
+	pass_flags = 0
+
+	tank_type = "tank"
+	water_capacity = 320		// Decent sized, holds almost 3 full buckets and a little more.
+	max_fish = 4				// Room for a few fish
+
+	has_lid = 1
+	max_integrity = 50				// Average strength, will take a couple hits from a toolbox.
+	shard_count = 2
+
+
+/obj/machinery/fishtank/wall
+	name = "wall aquarium"
+	desc = "This aquarium is massive! It completely occupies the same space as a wall, and looks very sturdy too!"
+	icon_state = "wall1"
+	density = 1
+	anchored = 1
+	pass_flags = 0				// This thing literally a fragile fish-wall, you can't throw past it.
+
+	tank_type = "wall"
+	water_capacity = 800		// This thing fills an entire tile, it holds a lot.
+	max_fish = 10				// Plenty of room for a lot of fish
+
+	has_lid = 1
+	max_integrity = 100			// This thing is a freaking wall, it can handle abuse.
+	shard_count = 3
+
+
+//////////////////////////////
+//		VERBS & PROCS		//
+//////////////////////////////
+
+/obj/machinery/fishtank/verb/toggle_lid_verb()
+	set name = "Toggle Tank Lid"
+	set category = "Object"
+	set src in view(1)
+	toggle_lid(usr)
+
+/obj/machinery/fishtank/proc/toggle_lid(var/mob/living/user)
+	lid_switch = !lid_switch
+	update_icon()
+
+/obj/machinery/fishtank/verb/toggle_light_verb()
+	set name = "Toggle Tank Light"
+	set category = "Object"
+	set src in view(1)
+	toggle_light(usr)
+
+/obj/machinery/fishtank/proc/toggle_light(var/mob/living/user)
+	light_switch = !light_switch
+	if(light_switch)
+		set_light(2,2,"#a0a080")
+	else
+		adjust_tank_light()
+
+//////////////////////////////
+//		NEW() PROCS			//
+//////////////////////////////
+
+/obj/machinery/fishtank/Initialize()
+	. = ..()
+	fish_list = new/list()
+	egg_list = new/list()
+	if(!has_lid)				//Tank doesn't have a lid/light, remove the verbs for then
+		verbs -= /obj/machinery/fishtank/verb/toggle_lid_verb
+		verbs -= /obj/machinery/fishtank/verb/toggle_light_verb
+
+/obj/machinery/fishtank/tank/Initialize()
+	. = ..()
+	if(prob(5))					//5% chance to get the castle decoration
+		icon_state = "tank2"
+
+//////////////////////////////
+//		ICON PROCS			//
+//////////////////////////////
+
+/obj/machinery/fishtank/update_icon()
+	overlays.Cut()
```

Comment:
no direct accessing overlays, use add_overlay()
additionally overlays are not text they're either images or mutable appearances last i checked